### PR TITLE
[BUGFIX] Update DefaultExpectationConfigurationBuilder to Handle All Parameters Of ExpectationConfiguration Properly

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Notes
* This enables the `meta` to be part of `ExpectationConfiguration` (in fact, all arguments are now enabled)
* @anthonyburdi Please see my response to your question (line 109 in `tests/rule_based_profiler/alice_user_workflow_verbose_profiler_config.yml`) -- I would like to remove up as many "TODO" items as possible (ideally with additional examples), so let us discuss.  Thank you!
* Test fixtures were updated (note that the expected `meta` did not match the configured -- small fix was made).

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [ENHANCEMENT], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
-
-
-

After submitting your PR, CI checks will run and @tiny-tim-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

Thank you for submitting!

### CODE REVIEW

File: great_expectations/rule_based_profiler/expectation_configuration_builder/default_expectation_configuration_builder.py
Comment (anthonyburdi) [inline great_expectations/rule_based_profiler/expectation_configuration_builder/default_expectation_configuration_builder.py:35]: Aha! Very nice fix 😄

File: great_expectations/rule_based_profiler/expectation_configuration_builder/default_expectation_configuration_builder.py
Comment (anthonyburdi) [inline great_expectations/rule_based_profiler/expectation_configuration_builder/default_expectation_configuration_builder.py:24]: Interesting - so the user can pass in a dict here which will get merged with any kwargs passed in. Is the idea that the user can be more explicit in config using this interface (vs kwargs)?

File: tests/rule_based_profiler/alice_user_workflow_fixture.py
Comment (anthonyburdi) [inline tests/rule_based_profiler/alice_user_workflow_fixture.py:131]: Good catch! ❤️

File: tests/rule_based_profiler/alice_user_workflow_verbose_profiler_config.yml
Comment (anthonyburdi) [inline tests/rule_based_profiler/alice_user_workflow_verbose_profiler_config.yml:109]: I think so! We did something similar in the Bob config: https://github.com/great-expectations/great_expectations/blob/0efe6c261bfd11f45f104b6b35f987a058c50cff/tests/rule_based_profiler/bob_user_workflow_verbose_profiler_config.yml#L107-L113

**Review by anthonyburdi — APPROVED** [review]
LGTM! Not sure if you want to address the question on line 109 before merging this or do that in a separate PR but approving in case of the latter.

File: great_expectations/rule_based_profiler/expectation_configuration_builder/default_expectation_configuration_builder.py
Comment (talagluck) [reply great_expectations/rule_based_profiler/expectation_configuration_builder/default_expectation_configuration_builder.py:35]: Just curious - is there a reason for this pattern vs. making `{}` the default value for meta on line 25?

File: great_expectations/rule_based_profiler/expectation_configuration_builder/default_expectation_configuration_builder.py
Comment (alexsherstinsky) [reply great_expectations/rule_based_profiler/expectation_configuration_builder/default_expectation_configuration_builder.py:24]: @anthonyburdi It is simpler -- once I realized that all we had to do was to match the signature of the constructor of `ExpectationConfiguration` -- once that is done, everything else falls into place (as long as the Rule's YML follows suite; otherwise, standard errors will be shown, as appropriate).

File: great_expectations/rule_based_profiler/profiler.py
Comment (talagluck) [inline great_expectations/rule_based_profiler/profiler.py:87]: Why do we get rid of this?

File: tests/rule_based_profiler/alice_user_workflow_verbose_profiler_config.yml
Comment (talagluck) [reply tests/rule_based_profiler/alice_user_workflow_verbose_profiler_config.yml:109]: Just confirming - I believe this is another piece of functionality that relies on active_batch, or else we wouldn't know where to look for that column.

**Review by talagluck — APPROVED** [review]

File: great_expectations/rule_based_profiler/expectation_configuration_builder/default_expectation_configuration_builder.py
Comment (anthonyburdi) [reply great_expectations/rule_based_profiler/expectation_configuration_builder/default_expectation_configuration_builder.py:35]: Yes @talagluck ! It's one of the Python gotchas (I learned from experience by screwing this up). If you instantiate the dict on line 25, it will be created when the class is first initialized, then every instance will mutate the same dict (likely will lead to strange errors). Here we are creating the dict in the constructor, e.g. one dict for each instance.

File: tests/rule_based_profiler/alice_user_workflow_verbose_profiler_config.yml
Comment (anthonyburdi) [reply tests/rule_based_profiler/alice_user_workflow_verbose_profiler_config.yml:109]: Good point. This assumes we look for the domain during the currently processed batch. I'm assuming that is done in an outer loop with an inner loop processing each domain in the currently in-process batch. Is that what you are thinking @talagluck ?

File: great_expectations/rule_based_profiler/expectation_configuration_builder/default_expectation_configuration_builder.py
Comment (talagluck) [reply great_expectations/rule_based_profiler/expectation_configuration_builder/default_expectation_configuration_builder.py:35]: Ah - makes sense! I think I remember discussing this with you before, but I must not have fully processed it.

File: tests/rule_based_profiler/alice_user_workflow_verbose_profiler_config.yml
Comment (talagluck) [reply tests/rule_based_profiler/alice_user_workflow_verbose_profiler_config.yml:109]: I think that makes sense. This is where I would think it would be possible to pass in other domain_kwargs, like `table` or `batch_id`, but where if we don't pass these in, we assume them from the active_batch. That's what I would expect, but I'm having trouble imagining if that is how this would actually work.

**Pull request merged**
